### DISCUSSION TITLE
AC and G-helper power mode difference?

### DISCUSSION THREAD

**Discussion by chockokruasan** [seerge/g-helper#2456] [Q&A]
Hello, I'm not sure is this issue or not, so I posted it here.
I have Asus Zephyrus duo 16 GX650PZ.

I want to migrate from AC to G-helper and use the same modes like in AC "silent" "balanced" and "turbo" without any tunning or tweaks. I have read there are no difference in modes beetwen AC and G-helper because they are built in bios. **BUT G-helper pulls dGPU TDP on each mode to the MAX, but I want to leave it as it was in AC.**

For example I checked performance in RDR2 benchmark with "turbo" mode with AC and G-helper 10 runs:
AC in "turbo" mode pulls to dGPU **155w** max and for CPU 40-45w during the test.
G-helper in "turbo" mode pulls to dGPU **175w** max and for CPU 40-45w

Why G-helper pulls dynamic boost by default? Can you make it off by default?

**And another strange issue:**
If I want to check performance with AC and G-helper in absolutly exact settings I set dGPU dynamic boost from 25w to 5w to make it the same like in AC (155w max) my CPU wattage with G-helper in the same test pulls up to **50-60w** (40-45w with AC) and this is increase temps up to 98. Why this happens?

Windows 11 23H2, AC uninstalled by Asus tool, all ASUS services stopped by G-helper.

**UPDATED**
This is not Dynamic Boost. G-helper set GPU TDP to MAX on all power modes.
video https://youtu.be/CUuj-Xoj0lY

**Comment by seerge** [comment]
@chockokruasan you can always set dynamic boost to 5W to match AC. 

If AC has 5w boost by default in Turbo, it looks more like a bug in AC itself, as Turbo supposed to deliver maximum performance.

If you want to limit CPU power it's also already possible

**Reply by chockokruasan** [reply]
Yes, I know. But I want to use defaults like in AC. Why G-helper default modes already tunned? Can you make it default and let me make any tweaks optionally?
That mean "silent" mode in G-helper is not so silent because it pulls dynamic boost to dGPU max by default and when I play games in silent mode my dGPU temps and fans works on max, right? If yes, this is not built in bios "silent" mode. It's already "custom".

"you can always set dynamic boost to 5W to match AC." 
When I set dynamic boost to 5W to match AC, G-helper pulls more CPU W than AC in same test. Checked 10 times. Why this happens?
Thanks!

**Reply by seerge** [reply]
Defaults are not tuned.

App just sets mode ID with a single command.

**Reply by chockokruasan** [reply]
CPU default, but dGPU not default, right?

**Reply by chockokruasan** [reply]
"If AC has 5w boost by default in Turbo, it looks more like a bug in AC itself, as Turbo supposed to deliver maximum performance."
Not certainly in that way. AC has about 60w CPU and 90w dGPU in "silent", 90w CPU and 125w dGPU in "balanced", 125w CPU and 150w+5w boost in "turbo" and you can add up to 130w CPU and 25w + OC dGPU in "custom". 
Can't understand is G-Helper pulls up max dynamic boost to dGPU for all modes, or all these modes built in bios with max dynamic boost already and AC set it lower for each mode?

**Comment by chockokruasan** [comment]
I will try to record a video with issue, this will make it more clear.

**Comment by chockokruasan** [comment]
@seerge 
Uploaded video
https://youtu.be/CUuj-Xoj0lY

**Comment by Kikimahe** [comment]
> Yes, the CPU and GPU TDP is different depending on the mode used, but G-helper changes my GPU TDP to the max on all modes, Is it only on my model? :) video https://youtu.be/CUuj-Xoj0lY

It's very curious indeed.
I would like my GPU to try to keep its maximum power in all modes. :)

On [your model](https://rog.asus.com/fr/laptops/rog-zephyrus/rog-zephyrus-duo-16-2023-series/spec/), I see that your GPU works in 2 ways:

- 140W+15W in dynamic boost mode (155W)
- 150W+25W in manual mode (175W)

It looks like from the video that your GPU can't stabilize on either one.
You are in CPU Balanced mode, So Your GPU should reach 175W in turbo mode only and not in Balanced mode.
Is it the same in Turbo Mode ?

It would also be weird if the problem came from Ghelper.
Have you tried restarting Windows without Ghelper at startup and setting it to Balanced mode in Windows directly?
Is it the same thing in HWinfo ?

![Capture d'écran 2023-12-08 225815](https://github.com/seerge/g-helper/assets/142881210/35fa0f0e-7c12-473f-bc62-171e6ecbbd4a)

**Reply by chockokruasan** [reply]
@Kikimahe

> I would like my GPU to try to keep its maximum power in all modes. :)

This is not correct because power based on cooling on each mode like @IceStormNG explained. If you will get max power on silent for example you will get overheating on GPU or this "silent" mode become "turbo" mode.

> It looks like from the video that your GPU can't stabilize on either one.

No, my GPU TDP is correct on each power mode when I click "Factory Defaults" in G-helper. But when I change something in G-helper, it is push MAX TGP on all modes.

**Comment by chockokruasan** [comment]
I tried to dig deeper into the G-helper log file and find out what happens.
When I click "Factory Defaults" on "Balanced" mode:

> 21.04.2024 15:52:50: Mode = 0 : OK
21.04.2024 15:52:50: ReadGPUPowerBase: -65536
21.04.2024 15:52:50: PowerLimit C0 (GPU BOOST) = 25 : OK
21.04.2024 15:52:50: PowerLimit C2 (GPU TEMP) = 87 : OK
21.04.2024 15:52:50: GET CLOCK LIMIT: 0
21.04.2024 15:52:50: GET GPU CLOCKS: 0, 0
21.04.2024 15:52:50: GET GPU CLOCKS: 0, 0
21.04.2024 15:52:50: GET CLOCK LIMIT: 0
21.04.2024 15:52:50: ReadGPUPowerBase: -65536

The comand "ReadGPUPowerBase: -65536" reapeat in the end, so I think it is load defaults from BIOS after all previous comands for GPU settings. Thats why it works normal when I click "Factory Defaults" every time.

But when I switch for example to "silent" and back to "balanced" setting from "GPU" section overwrite default GPU TPD to the MAX?

> 21.04.2024 16:03:24: Power Mode 00000000-0000-0000-0000-000000000000 -> 961cc777-2547-4f9d-8174-7d86181b8a7a:OK
21.04.2024 16:03:24: Boost 0
21.04.2024 16:03:24: Mode = 2 : OK
21.04.2024 16:03:24: GET GPU CLOCKS: 0, 0
21.04.2024 16:03:24: GET CLOCK LIMIT: 0
21.04.2024 16:03:24: ReadGPUPowerBase: -65536
21.04.2024 16:03:25: Power Mode 961cc777-2547-4f9d-8174-7d86181b8a7a -> 00000000-0000-0000-0000-000000000000:OK
21.04.2024 16:03:25: Boost 2
21.04.2024 16:03:25: Mode = 0 : OK
21.04.2024 16:03:25: GET GPU CLOCKS: 0, 0
21.04.2024 16:03:25: GET CLOCK LIMIT: 0
21.04.2024 16:03:25: ReadGPUPowerBase: -65536
21.04.2024 16:03:26: PowerLimit C0 (GPU BOOST) = 25 : OK
21.04.2024 16:03:26: PowerLimit C2 (GPU TEMP) = 87 : OK
21.04.2024 16:03:27: PowerLimit C0 (GPU BOOST) = 25 : OK
21.04.2024 16:03:27: PowerLimit C2 (GPU TEMP) = 87 : OK

**Reply by seerge** [reply]
> The comand "ReadGPUPowerBase: -65536" reapeat in the end, so I think it is load defaults from BIOS after all previous comands for GPU settings. Thats why it works normal when I click "Factory Defaults" every time.

That's a check if device has power limits from 2024 year models. It has no effect in your case, also you can't read any settings from BIOS.

> 21.04.2024 16:03:26: PowerLimit C0 (GPU BOOST) = 25 : OK
> 21.04.2024 16:03:26: PowerLimit C2 (GPU TEMP) = 87 : OK

It sets Boost and Temp limits cause this how it's set with corresponding sliders in GPU section.

I can make that "Factory Defaults" will completely erase that setting and won't set it anymore unless you explicitly drag a slider.

**But I believe that's a bad idea, as slider position will not match actual setting in the UI.**

**Comment by chockokruasan** [comment]
@seerge hello, did you see my video? Can you comment why I have this issue? Something wrong with my laptop?) Looks like other users or maybe other laptops do not have this issue. @Kikimahe has different TDP GPU on each modes.

> I can make that "Factory Defaults" will completely erase that setting and won't set it anymore unless you explicitly drag a slider.

> But I believe that's a bad idea, as slider position will not match actual setting in the UI.

Can you make maybe "Apply" checkbox under GPU setting like in CPU settings?
![111111111](https://github.com/seerge/g-helper/assets/166668221/a940ccb1-6750-472a-9103-241a578c3c8e)

**Reply by seerge** [reply]
@chockokruasan for GPU settings not everything needs to be "applied". For example clocks and clock limits have immediate effect and also reflect actual state (as app gets current values directly from nvidia API). So i don't want to overcomplicate things here.

There is nothing wrong with your laptop. But it could be quite unique situation when in your case Turbo mode doesn't have maximum Dynamic Boost setting in AC (while all other models seem to use maximum possible boost setting).

So if you want (by some reason) reflect same settings as you had in AC, just set boost to 5W and you will be fine.

**Reply by seerge** [reply]
@chockokruasan @Kikimahe 

If you want you can try this build. 

I have adjusted ``Factory Defaults`` button behavior a bit. Instead of resetting ``Dynamic Boost`` to 25W and ``Temp Limit`` to 87C it will erase values completely from settings, and won't apply them until you explicitly interact with a slider. 

This should allow you to use Silent mode for example without any customizations at all.

**However I'm not sure if this is a good UI solution.** As in UI boost slider will still show 25W.

[GHelper.zip](https://github.com/seerge/g-helper/files/15052862/GHelper.zip)

**Reply by Kikimahe** [reply]
Sorry Seerge, but I don't have any problem with my GPU's TDP. I don't need a fix. :)

**Reply by chockokruasan** [reply]
@seerge At 1st many thanks for your time and your app.
But it seems we're talking about different things :) Let's forget about dynamic boost, it works as it should I think.

G-helper apply factory defaults for each power mode correctly!!! **The problem is** if I touch anything after default applied in G-helper it push GPU power to the MAX for all modes even if I just switch between modes (not dynamic boost, maybe clock limit or clock offset I dont know). Every power mode by default based on cooling. For example "balanced" mode has 90w max power to CPU and 130w power for GPU and medium fan speed, but G-helper push GPU (only) power to MAX 175w and its overheat GPU.

I uploaded video on youtube where I clearly show the problem. https://youtu.be/CUuj-Xoj0lY Please watch it (3 mins). it is hard to explain by text.

@Kikimahe and many other users don't have this problem.

Many thanks again!

**Reply by seerge** [reply]
@chockokruasan as soon as you apply even some custom settings (i.e. drag a slider) - BIOS may treat that as a Manual mode and crank up the limits. This is how it’s done by asus for your specific model probably

**Reply by chockokruasan** [reply]
@seerge did you watch video? It happens even If I change power modes with no apply and no changes. I dont have this problem with AC, only in G-helper.
 And it touch only GPU power, CPU is the same.
Thats why I offered to add "Apply" checkbox for GPU Clock Core and Clock offset in GPU setting like in CPU power settings. Why do you think this is not good UI solution? Looks like this is pretty the same when we need to click "Apply" checkbox in CPU power settings and GPU power settings? By the way in AC we also need to click "apply" in custom mode GPU and CPU section.

**Reply by seerge** [reply]
@chockokruasan when you set the mode, it would also set previously saved Boost/Temp values and (probably) trigger a "manual" mode in your specific case.

```
21.04.2024 16:03:25: Mode = 0 : OK <- balanced mode
21.04.2024 16:03:26: PowerLimit C0 (GPU BOOST) = 25 : OK <--- as soon as at least something is set, it may treat it as "manual"
21.04.2024 16:03:26: PowerLimit C2 (GPU TEMP) = 87 : OK
```

That's why I linked a build that would  erase saved values after clicking  ``Factory Defaults`` and not apply them anymore (unless you explicitly interact with a slider)

**Reply by chockokruasan** [reply]
@seerge Thanks, your linked build here is work! I can use now default Asus modes. Can you add maybe in Main build in the next updates option with checkbox "Apply power limits" for GPU settings or option to activate this by command in config.json like other things like language, optional UV levels etc.? Thanks!
